packages/gossip/manager_test.go

```diff
@@ -372,6 +372,46 @@ func TestTxRequest(t *testing.T) {
 	e.AssertExpectations(t)
 }
 
+func TestDropNeighbor(t *testing.T) {
+	mgrA, closeA, peerA := newTestManager(t, "A")
+	defer closeA()
+	mgrB, closeB, peerB := newTestManager(t, "B")
+	defer closeB()
+
+	connect := func() {
+		var wg sync.WaitGroup
+		closure := events.NewClosure(func(_ *Neighbor) { wg.Done() })
+		wg.Add(2)
+		Events.NeighborAdded.Attach(closure)
```

Comment:
I believe that `Events` should be part of the actual neighbor manager struct instead of being solely defined in the gossip package.

Reply:
I agree 100%, but I think this is out of the scope of this simple PR.
I'll add an issue to do that for gossip and autopeering.

Reply:
#349